File: easybluetoothble/src/main/java/com/srise/easybluetoothble/Interface/ISendDataEngine.java
package com.srise.easybluetoothble.Interface;

import com.srise.easybluetoothble.bean.BaseMessage;

import java.util.List;

public interface ISendDataEngine {
    /**
     * 根据情况分包
     *
     * @param data
     * @return
     */

    List<byte[]> splitDataPacket(byte[] data);


    /**
     * 发送数据
     *
     * @param bleSendMessage
     * @return
     */
     void sendData(BaseMessage bleSendMessage);
}

File: ktlionble/src/main/java/c/example/dell/ktlionble/inter/ISendDataEngine.kt
package c.example.dell.ktlionble.inter

import c.example.dell.ktlionble.bean.BaseMessage

interface ISendDataEngine {
    /**
     * 根据情况分包
     *
     * @param data
     * @return
     */
    fun splitDataPacket(data: ByteArray?): List<ByteArray?>?


    /**
     * 发送数据
     *
     * @param bleSendMessage
     * @return
     */
    fun sendData(bleSendMessage: BaseMessage)
}